deltachat-ffi/deltachat.h

```diff
@@ -986,6 +986,9 @@ uint32_t        dc_add_device_msg            (dc_context_t* context, const char*
 
 
 /**
+ * DEPRECATED This is now done automatically while configuring.
```

Comment:
Currently android code calls it on every creation of chatlist.

If it shouldn't do it, then this function is better removed completely, otherwise everyone will forget to remove calls to it.

I don't see any problems with it, except that imported backups from old versions of DC will not get a welcome message. It is even better this way.

After that, maybe "self-chat-added" check and config value could also be removed, since the function is only called once?

Reply:
> After that, maybe "self-chat-added" check and config value could also be removed, since the function is only called once?

Problem is, configure can be started multiple times (because it fails the first time) and this fn will be called everytime.

> If it shouldn't do it, then this function is better removed completely, otherwise everyone will forget to remove calls to it.

@hpk42 said that we should not break backwards compability where possible (@hpk42 correct?), that's why I didn't up to now. At least compilation will fail but I don't know how obvious it will be that the function was removed.

What I can do is make the body of this fn empty because it does nothing anyway and make update_device_chats() `pub(crate)`.

Reply:
Desktop currently calls this function on each login.
Anyways I agree with @link2xt this time.
Normally I would suggest marking this function as `@deprecated` and to turn it into a no-op, but it looks like we have some breaking changes* in the next version anyway so I believe it makes sense to already remove it now.
(given that we don't forget to mention it in the changelog)
*some PRs that indroduce breaking changes, too: (https://github.com/deltachat/deltachat-core-rust/pull/1968 https://github.com/deltachat/deltachat-core-rust/pull/1965)

Reply:
Ok, I'll do that as soon as we decided what to do about bots.